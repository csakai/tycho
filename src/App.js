import React from 'react';
import React3 from 'react-three-renderer';
import * as THREE from 'three';
import ReactDOM from 'react-dom';
import Orbital from './components/Scene/Objects/Orbital';
import data from './global/fixtures';
import Controls from './components/Scene/Utils/Controls';
import Clock from './engine/clock';
import Label from './components/Label/Label';

class App extends React.Component {
  constructor(props, context) {
    super(props, context);

    // construct the position vector here, because if we use 'new' within render,
    // React will think that things have changed when they have not.
    this.cameraPosition = new THREE.Vector3(300, 300, 300);
    this.state = {positions: {}};
    this.clock = new Clock();
    this.clock.speed(4);
  }

  componentDidMount = () => {
    this.controls = new Controls(this.refs.camera);
  }

  componentWillUnmount = () => {
    this.controls.dispose();
    delete this.controls;
  }

  _onAnimate = () => {
    if (this.clock) {
      this.setState({
        time: this.clock.getTime()
      });
    }
  }

  updateScreenPositions = (position, id) => {
    this.setState({
      positions: Object.assign(this.state.positions, {
        [id]: position
      })
    });
  }

  getOrbitalElements = (orbitals, odd) => {
    return orbitals.map((orbital) => (
      <Orbital {...orbital}
        time={this.state.time}
        camera={this.refs.camera}
        onUpdate={this.updateScreenPositions}
        odd={odd}
        key={orbital.id}>
        {orbital.children && this.getOrbitalElements(orbital.children, !odd)}
      </Orbital>
    ));
  }

  getOrbitalLabels = (orbitals) => {
    let labels = [];

    orbitals.forEach((orbital) => {
      labels.push(
        <Label
          position={this.state.positions[orbital.id]}
          orbital={orbital}
          key={orbital.id}>
          {orbital.children && this.getOrbitalLabels(orbital.children)}
        </Label>
      );

      if (orbital.children) {
        labels = labels.concat(this.getOrbitalLabels(orbital.children));
      }
    });
    return labels;
  }

  render() {
    const width = window.innerWidth; // canvas width
    const height = window.innerHeight; // canvas height

    return (
      <div>
        <React3
          mainCamera="camera" // this points to the perspectiveCamera which has the name set to "camera" below
          width={width}
          height={height}
          antialias={true}
          onAnimate={this._onAnimate}
          alpha={true}>
          <scene>
            <perspectiveCamera
              name="camera"
              ref="camera"
              fov={50}
              aspect={width / height}
              near={1}
              far={10000}
              position={this.cameraPosition}
            />
            {this.getOrbitalElements(data)}
            <axisHelper size={500} />
          </scene>
        </React3>
        {this.getOrbitalLabels(data)}
      </div>
    );
  }
}

ReactDOM.render(<App />, document.getElementById('root'));
