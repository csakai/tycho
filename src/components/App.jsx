import React from 'react';
import data from '../global/fixtures';
import Clock from '../utils/Clock';
import SceneContainer from '../containers/SceneContainer';
import LabelGroup from './LabelGroup';
import UIControlsContainer from '../containers/UIControlsContainer';

export default class App extends React.Component {

  componentWillMount = () => {
    this.clock = new Clock();
    this.state = {
      positions: {},
      time: this.clock.getTime()
    };
    this.clock.speed(5); // temporary
  }

  onAnimate = () => {
    this.setState({
      time: this.clock.getTime()
    });
  }

  updateScreenPosition = (position, id) => {
    this.setState({
      positions: Object.assign(this.state.positions, {
        [id]: position
      })
    });
  }

  render() {
    return (
      <div>
        <SceneContainer
          onAnimate={this.onAnimate}
          updateScreenPosition={this.updateScreenPosition}
          time={this.state.time}
          orbitalData={data}
          width={window.innerWidth}
          height={window.innerHeight}
          perspective={false}
          targetName="dummyOuter"
          lookAtName="dummyParent"
        />
        <LabelGroup
          positions={this.state.positions}
          orbitalData={data}
        />
        <UIControlsContainer />
      </div>
    );
  }
}
