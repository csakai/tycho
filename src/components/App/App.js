import React from 'react';
import data from '../../global/fixtures';
import Clock from '../../utils/Clock';
import Scene from '../Scene';
import LabelGroup from '../LabelGroup';

export default class App extends React.Component {

  componentWillMount = () => {
    this.clock = new Clock();
    this.state = {
      positions: {},
      time: this.clock.getTime()
    };
    this.clock.speed(4); // temporary
  }

  onAnimate = () => {
    this.setState({
      time: this.clock.getTime()
    });
  }

  updateScreenPositions = (position, id) => {
    this.setState({
      positions: Object.assign(this.state.positions, {
        [id]: position
      })
    });
  }

  render() {
    return (
      <div>
        <Scene
          onAnimate={this.onAnimate}
          updateScreenPositions={this.updateScreenPositions}
          time={this.state.time}
          orbitalData={data}
        />
        <LabelGroup
          positions={this.state.positions}
          orbitalData={data}
        />
      </div>
    );
  }
}
