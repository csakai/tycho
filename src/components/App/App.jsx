import React from 'react';
import SceneContainer from '../../containers/SceneContainer';
import UIControlsContainer from '../../containers/UIControlsContainer';
import LoaderContainer from '../../containers/LoaderContainer';
import TourContainer from '../../containers/TourContainer';
import ModalContainer from '../../containers/ModalContainer';
import StatsContainer from '../../containers/StatsContainer';
import Constants from '../../constants';

export default class App extends React.Component {
  render() {
    return (
      <div>
        <SceneContainer
          onAnimate={this.props.onAnimate}
          width={window.innerWidth}
          height={window.innerHeight}
        />
        <UIControlsContainer />
        <LoaderContainer />
        <TourContainer labels={Constants.Tour.LABELS} />
        <ModalContainer
          type={Constants.UI.ModalTypes.STATS_MODAL}
          title={this.props.title}>
          <StatsContainer />
        </ModalContainer>

      </div>
    );
  }
}


