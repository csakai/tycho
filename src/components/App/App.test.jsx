import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import App from './App';

describe('App Component', () => {
    let component;

    beforeEach(() => {
        component = shallow(<App
            onAnimate={jest.fn()}
            updateScreenPosition={jest.fn()}
            time={1}
            width={300}
            orbitalData={[]}
            positions={{}}
            height={500}
            pageText={{
                aboutTitle: 'About',
                aboutInfo: 'About this project'
            }}
        />);
    });

    describe('render()', () => {
        it('should render the app successfully', () => {
            expect(toJson(component)).toMatchSnapshot();
        });
    });
});
