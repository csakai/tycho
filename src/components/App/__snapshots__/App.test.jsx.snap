// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component render() should render the app successfully 1`] = `
<div>
  <Connect(SceneContainer)
    height={768}
    onAnimate={[Function]}
    width={1024}
  />
  <Connect(UIControlsContainer) />
  <Connect(LoaderContainer) />
  <Connect(TourContainer)
    labels={
      Array [
        Object {
          "duration": 1000,
          "text": "Welcome to the Solar System",
        },
        Object {
          "duration": 1000,
          "text": "This is a real-time interactive simulation of major planetary bodies",
        },
        Object {
          "duration": 1000,
          "text": "Let's start exploring",
        },
      ]
    }
  />
  <Connect(ModalContainer)
    type="STATS_MODAL"
  >
    <Connect(StatsContainer) />
  </Connect(ModalContainer)>
</div>
`;
